src/Compilers/CSharp/Portable/CodeGen/CodeGenerator.cs

```diff
@@ -102,12 +131,17 @@ private enum IndirectReturnState : byte
 
             _boundBody = Optimizer.Optimize(
                 boundBody, 
-                debugFriendly: _optimizations != OptimizationLevel.Release, 
+                debugFriendly: _ilEmitStyle != ILEmitStyle.Release, 
                 stackLocals: out _stackLocals);
 
             _methodBodySyntaxOpt = (method as SourceMethodSymbol)?.BodySyntax;
         }
 
+        private bool IsDebugPlus()
+        {
+            return this._module.Compilation.Options.DebugPlusMode;
```

Comment:
I tried that and it gets complicated. 
It is not yet clear whether this functionality needs to be a part of public APIs for compilation or emit.

Passing the flag as an internal flag on a compilation allows to defer that decision. 
For now it is more like a extendedCustomDebugInformation - an internal flag that is somewhat hard to 
conceptualize into public API and may not be worth the additional clutter.